options/options.go

```diff
@@ -452,6 +461,60 @@ func (opts *ToolOptions) ParseArgs(args []string) ([]string, error) {
 	return args, err
 }
 
+// ParseConfigFile iterates over args to find a --config option. If not found, we return.
+// If found, we read the contents of the specified config file in YAML format. We parse
+// any values corresponding to --password, --uri and --sslPEMKeyPassword, and store them
+// in the opts.
+func (opts *ToolOptions) ParseConfigFile(args []string) error {
+	// Get config file path from the arguments, if specified.
+	configPath := ""
+	for i, arg := range args {
+		if strings.HasPrefix(arg, "--config=") {
+			// No value after equal sign.
+			if len(arg) < 10 {
+				return fmt.Errorf("--config option specified but no file given")
+			}
+			configPath = arg[9:]
+			break
+		} else if strings.HasPrefix(arg, "--config") {
+			// No next argument or the next argument is an option.
+			if i+1 == len(args) || strings.HasPrefix(args[i+1], "-") {
```

Comment:
One ambiguity I ran into was if the user does this:

`$ mongodump --config -d myDB -c ...`

Do we want to treat this as a missing config file, or try to open a file named `-d` instead? I see validity in both, though I'm maybe leaning a bit towards the latter option of just trying to open it and letting the user deduce what happened from the `os.Open()` error message.

Reason being that they could theoretically have a config file that starts with a hyphen:

`$mongodump --config -tools-config.yaml -d myDB -c ...`

It's unlikely, but we'd either have to ensure the arg after `--config` isn't an option (lot of manual checking) or force the user to rename `-tools-config.yaml` to something else.

I can remove the `|| strings.HasPrefix(args[i+1], "-")` check if others agree.

Reply:
Yeah good question, `go-flags` has this behavior:

```
$ mongodump --sslPEMKeyFile -d
2020-11-25T18:58:47.875+0100    error parsing command line options: expected argument for flag `--sslPEMKeyFile', but got option `-d'
```

So I think disallowing files beginning with `-` is consistent. If people do have files beginning with a `-`, they can always use the `--config="foo"` format. But I also think trying to open `-d` is fine too. It would very quickly obvious what the problem is. If you prefer that approach, I think go for it.

Reply:
Cool, I'm still leaning towards trying to open `-d` just because otherwise the error message would be kinda hard to get right. Like if a user says `--config -tools-config.yaml` the error would be `--config option specified but no file given`, which would be confusing since they obviously did provide a file and might not think to use `--config=`. We could separate out the conditions, but that's already more complexity than it's worth I feel.

Instead I'll add a bit of context to the `ioutil.ReadFile()` error, so now it'll look like:
```
$ mongodump --config -d myDB
2020-11-25T14:26:20.168-0500	error parsing command line options: error opening file with --config: open -d: no such file or directory
```

If it's at the end, it'll stay the same:
```
$ mongodump -d myDB --config
2020-11-25T14:35:30.196-0500	error parsing command line options: --config option specified but no file given
```